src/test/lrsql/util/reaction_test.clj

```diff
@@ -90,3 +90,76 @@
+                 "verb"   {"id" "https://example.com/verbs/completed"}
+                 "object" {"id"         "https://example.com/activities/a"
+                           "objectType" "Activity"}
+                 "result" [{"success" true} {"success" false}]}
+   :completed_b {"id"     "c51d1628-ae4a-449f-8d8d-13d57207f468"
+                 "actor"  {"mbox" "mailto:bob@example.com"}
+                 "verb"   {"id" "https://example.com/verbs/completed"}
+                 "object" {"id"         "https://example.com/activities/b"
+                           "objectType" "Activity"}
+                 "result" [{"success" false} {"success" true}]}})
+
+(deftest statement-generation-test
+  (testing "Basic statement generation"
+    (is (= (r/generate-statement
+            fake-cond-map
+            {"actor" {"mbox" {"$templatePath" ["completed_a" "actor" "mbox"]}}
+             "verb"  "did"
+             "object"  "this"})
+           {"actor" {"mbox" "mailto:bob@example.com"}
+            "verb"  "did"
+            "object"  "this"})))
+
+  (testing "Inclusion of multiple attributes"
+    (is (= (r/generate-statement
+            fake-cond-map
+            {"actor" {"mbox" "mailto:bob@example.com"
+                      "madeUpAttr" {"$templatePath" ["completed_a" "actor" "madeUpAttr"]}}
+             "verb"  "did"
+             "object"  "this"})
+           {"actor" {"mbox" "mailto:bob@example.com"
+                     "madeUpAttr" "8"}
+            "verb"  "did"
+            "object"  "this"})))
+  (testing "non-string contents"
+    (is (= (r/generate-statement
+            fake-cond-map
+            {"verb"   {"$templatePath" ["completed_b" "verb"]}})
+           {"verb"  {"id" "https://example.com/verbs/completed"}})))
+
+  (testing "generate statements patched together from different conditions"
+    (is (= (r/generate-statement
+            fake-cond-map
+            {"actor"  {"$templatePath" ["completed_a" "actor"]}
+             "verb"   "did"
+             "object" {"$templatePath" ["completed_b" "object"]}})
+
+           {"actor" {"mbox" "mailto:bob@example.com"
+                     "madeUpAttr" "8"}
+            "verb"  "did"
+            "object" {"id" "https://example.com/activities/b"
+                      "objectType" "Activity"}})))
+  (testing "access by index"
+    (is (= (r/generate-statement
+            fake-cond-map
+            {"a_first_result" {"$templatePath" ["completed_a" "result" 0 "success"]}
+             "b_first_result" {"$templatePath" ["completed_b" "result" 0 "success"]}})
+           {"a_first_result" true
+            "b_first_result" false})))
+
+  (testing "when no value at path, throws appropriate exception"
+    (is (= (try (r/generate-statement
+                  fake-cond-map
+                  {"actor_birthday" {"$templatePath" ["completed_a" "actor" "birthday"]}})
+
+                (catch clojure.lang.ExceptionInfo e (ex-info "No value found at [:completed_a \"actor\" \"birthday\"]"
```

Comment:
Two points of convention:
* We generally try to keep code to an 80 char horizontal max (this one is looking a little wide)
* We generally line break after the exception binding in `catch` block. Not something I even think about but all of our code thusfar has done it, lol.